source/adios2/engine/bp4/BP4Writer.cpp

```diff
@@ -201,6 +203,24 @@ void BP4Writer::InitParameters()
     m_DrainBB = m_WriteToBB && m_BP4Serializer.m_Parameters.BurstBufferDrain;
 }
 
+std::string scrRoute(std::string name)
+{
+    char scr_name[SCR_MAX_FILENAME];
+    SCR_Route_file(name.c_str(), scr_name);
```

Comment:
Where is `SCR_Route_file` define and what does it do? Is it changing the path where the files are stored and this is how SCR knows it needs to handle them?

Reply:
Right.  This is how SCR learns which files belong to a checkpoint, and it's how SCR tells the caller where the file should be opened instead of its normal location.

``SCR_Route_file`` is declared in ``scr.h`` and is implemented in ``scr.c``.

The first argument is an input parameter that specifies the path where the application wants to open its file.    This usually will be an absolute or relative path to some place on the parallel file system.  SCR records this final path in its internal data structures.

The second argument is an output parameter.  This is the temporary path that SCR provides back to the caller.  It's the path where the file should actually be opened.  For example, when using node-local SSDs, the path returned here will be prefixed with the SSD mount point.

Reply:
Thanks for the explanation. I will double check if you cover all cases, but at a fist glance this looks good.